### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by Sami-Ullah-Tufail**
### This pull request has been created against below mentioned new implementations into the project:

- [ ] Installed necessary dependencies for Devise.
- [ ]  Generated configuration files for Devise.
- [ ]  Configured default URL options for Devise.
- [ ]  Updated the User model using Devise.
- [ ]  Executed migration files for database setup.
- [ ]  Generated Devise views for user authentication.
- [ ]  Implemented functionality for user sign-up and sign-out.
- [ ]  Updated relevant view files.
- [ ]  Added some basic styling.
- [ ]

### CODE REVIEW

**Comment by Sami-Ullah-Tufail** [comment]
[
WhatsApp Image 2024-01-20 at 14 23 15
](url)

**Comment by Reem-lab** [comment]
@Sami-Ullah-Tufail  could you check your slack?

**Comment by Sami-Ullah-Tufail** [comment]
> @Sami-Ullah-Tufail could you check your slack?

checking

File: tmp/cache/assets/sprockets/v4.0.0/-3/-3CZ5hLd6mf3BRmYZdKnlWYUPmWER3btDw00bzAC7MM.cache
Comment (OyePriscilla) [inline tmp/cache/assets/sprockets/v4.0.0/-3/-3CZ5hLd6mf3BRmYZdKnlWYUPmWER3btDw00bzAC7MM.cache:1]: [Optional] Please kindly remove these extra thousands of files that are not relevant to your project at the moment. 👍

**Review by OyePriscilla — APPROVED** [review]
Hi @Sami-Ullah-Tufail 👍 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
Build a web app that requires the user to log in. ✔️ 
Use devise gem for authentication. ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**